tests/e2e/vsan_stretched_cluster.go

```diff
@@ -3290,4 +3290,680 @@ var _ = ginkgo.Describe("[vsan-stretch-vanilla] vsan stretched cluster tests", f
+				and wait for them to be bound
+			3.	Delete four PVCs with reclaim policy Retain
+			4.	Delete two PVs reclaim policy Retain related to PVC used in step 3
+			5.	Create two pods using PVCs with reclaim policy Delete
+			6.	Bring vsan-health service down
+			7.	Create two pods with two PVCs each
+			8.	Create two static PVs with disk left after step 4
+			9.	Create two PVCs to bind to PVs with reclaim policy Retain
+			10. Delete four PVCs with reclaim policy Retain different from the ones used in step 3 and 9
+			11. Delete two PVs reclaim policy Retain related to PVC used in step 10
+			12. Add labels to all PVs, PVCs
+			13. Bring vsan-health service up when full sync is triggered
+			14. Bring down primary site
+			15. Verify that the VMs on the primary site are started up on the other esx servers
+		        in the secondary site
+			16. Wait for full sync
+			17. Verify CNS entries
+			18. Delete all pods, PVCs and PVs
+			19. Bring primary site up and wait for testbed to be back to normal
+
+	*/
+	ginkgo.It("[primary-centric] Site failover during full sync", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating StorageClass")
+		// decide which test setup is available to run
+		ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+		scParameters = map[string]string{}
+		scParameters["StoragePolicyName"] = storagePolicyName
+		storageClassName = "nginx-sc-delete"
+		var pods []*v1.Pod
+		var pvclaimsWithDelete, pvclaimsWithRetain []*v1.PersistentVolumeClaim
+		var volHandles []string
+
+		scRetain, err := createStorageClass(client, scParameters, nil, v1.PersistentVolumeReclaimRetain, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		scSpec := getVSphereStorageClassSpec(storageClassName, scParameters, nil, "", "", false)
+		scDelete, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, scRetain.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = client.StorageV1().StorageClasses().Delete(ctx, scDelete.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		for i := 0; i < 6; i++ {
+			framework.Logf("Creating pvc %v with reclaim policy Delete", i)
+			pvc, err := createPVC(client, namespace, nil, diskSize, scDelete, "")
+			pvclaimsWithDelete = append(pvclaimsWithDelete, pvc)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		for i := 0; i < 8; i++ {
+			framework.Logf("Creating pvc %v with reclaim policy Retain", i)
+			pvc, err := createPVC(client, namespace, nil, diskSize, scRetain, "")
+			pvclaimsWithRetain = append(pvclaimsWithRetain, pvc)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		defer func() {
+			for _, pvclaim := range pvclaims {
```

Comment:
where is `pvclaims` declared? Also, volume withs Retain reclaim policy need to be cleanup properly.